x/twap/logic.go

```diff
@@ -243,11 +252,43 @@ func computeArithmeticTwap(startRecord types.TwapRecord, endRecord types.TwapRec
 		}
 		return endRecord.P1LastSpotPrice, err
 	}
+
+	if twapType == arithmeticTwapType {
+		return computeArithmeticTwap(startRecord, endRecord, quoteAsset), err
+	}
+	return computeGeometricTwap(startRecord, endRecord, quoteAsset), err
+}
+
```

Comment:
we should explicitly take twapType == geometricTwapType as and else if here. Otherwise we might accidentally compute geomtric twap for an invalid type input

Reply:
Good point. I changed `twapType` to be a boolean so this is not possible anymore